File: app/src/main/java/ch/bailu/aat/views/osm_features/OnSelected.java
package ch.bailu.aat.views.osm_features;

import ch.bailu.aat.util.filter_list.ListEntry;

public interface OnSelected {
    int EDIT = 0;
    int SHOW = 1;
    int FILTER = 2;
    int SELECT = 3;

    OnSelected NULL = (e, action, variant) -> {

    };


    void onSelected(ListEntry e, int action, String variant);
}

File: aat-android/src/main/kotlin/ch/bailu/aat/views/osm/features/OnSelected.kt
package ch.bailu.aat.views.osm.features

import ch.bailu.aat_lib.lib.filter_list.AbsListItem

fun interface OnSelected {
    enum class Action { Edit, Filter }

    fun onSelected(e: AbsListItem, action: Action, variant: String)

    companion object {

        @JvmField
        val NULL = OnSelected { _: AbsListItem, _: Action, _: String -> }
    }
}
